tests/include/test/arguments.h

```diff
@@ -0,0 +1,38 @@
+/**
+ * \file arguments.h
+ *
+ * \brief Manipulation of test arguments.
+ *
+ * Much of the code is in host_test.function, to be migrated here later.
+ */
+
+/*
+ *  Copyright The Mbed TLS Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#ifndef TEST_ARGUMENTS_H
+#define TEST_ARGUMENTS_H
+
+#include "mbedtls/build_info.h"
+#include <stdint.h>
+#include <stdlib.h>
+
+typedef union {
+    size_t len;
```

Comment:
Sorry I forgot this in the last review. 

The name `len` always makes me confuse. I was thinking it the length of `mbedtls_test_argument_t` and then realised it shouldn't be because `mbedtls_test_argument_t` is union type. 
So I think we could change it to a name that clearly indicates its a `type`. But I'm not sure which name is better, maybe `size_t` or `type_size`? What's your opinion?

Reply:
I'm sorry, I don't understand. At the moment, a value of type `mbedtls_test_argument_t` can be either the length of some string (then its type is `size_t` and the union field is called `len`), or a signed integer (then its type is `intmax_t` to support the widest possible range, and the field is called `sint`). A value of type `mbedtls_test_argument_t` is never a type indicator.

Note that in a follow-up, when the test framework supports arbitrary-sized unsigned integers, there will be a field of type `uintmax_t` which will replace the field of type `size_t`.

Reply:
I meant that the name `len` makes people (maybe only me) difficult to realise its type. But it is not so important becauce it's almost in generated files.

> Note that in a follow-up, when the test framework supports arbitrary-sized unsigned integers, there will be a field of type uintmax_t which will replace the field of type size_t.

If there is a plan to replace this field, I'm fine with the naming here for the time being. BTW, I think `uintmax_t` looks better than `size_t` here.